index.rst

```diff
@@ -1,60 +1,850 @@
+
+:tocdepth: 1
+
+.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+
+.. TODO: Delete the note below before merging new content to the master branch.
+
+.. note::
+
+   **This technote is not yet published.**
+
+   The ``pipetask`` tool included in the Gen3 middleware for running `PipelineTasks<lsst.pipe.base.PipelineTask>` is often opaque and hard to use.  This is partly the result of history; the original interface was much more straightforward, but it assumed a very different collections model than what Butler has now, and the changes made to adapt it to the new collections system did not go deep enough.
+   But it also suffers from the fact that many of its steps are "leaky"; information provided in one step (e.g. input collections used for `~lsst.pipe.base.QuantumGraph`) cannot be straightforwardly passed to later steps (e.g. execution of that `~lsst.pipe.base.QuantumGraph`), forcing it to rely on the user to provide consistent arguments and limiting the guarantees we can make about behavior in the presence of user error.
+   Finally, the ``pipetask`` interface is currently only really usable from the command-line, making it hard to write unit tests for concrete `PipelineTasks<lsst.pipe.base.PipelineTask>`; even if we were to keep the command-line interface the same, we need to refactor the implementation to provide a usable Python API.
+   This technote will provide a detailed design sketch for a new ``pipetask`` interface that attempts to address these problems, centered around a ``git``-like "campaign directory" concept that will be used to save state between different ``pipetask`` invocations.
```

Comment:
Trying to analogize to `git` is a bit dangerous, as there is no ability to recover previous campaign state in the commands as given, at least without manually re-creating it from externally persisted provenance.